### DISCUSSION TITLE
Setting granular group permissions not working

### DISCUSSION THREAD

**Discussion by artemis-mike** [requarks/wiki#4467] [Error / Bug Report]
# Problem
We're having a global group, which has some allow-rights on `/` 
Now there are some groups with different subsets of users, with each group having a separate area, that should only be accessible by users of that specific group.

Unfortunately, as we designed our ruleset, nobody is able to access the restricted subarea. Please see the example below. 

Did we design our rules wrong? Is this a bug? 

Even with one applicable `deny` overriding all `allow` matching the same path, there must be a way to have a global group of users, who have access to a large area of the wiki and using subgroups to allow access to otherwise restricted areas.

The [documentation](https://docs.requarks.io/groups) is not very clear in that point.
It states 

> **What is the default behavior for a permission?**
> 
> Unless you explicitely grant `Allow` on a permission, it will always be denied by default. Therefore, giving no permission is the same as adding a `Deny` rule for all permissions. As such, `Deny` rules are only needed to override a previous `Allow` rule. You do not need to add a `Deny` rule if you didn't `Allow` it in the first place at a lower level.

But in this usecase I want to use a positive-list with everything being accessible and restricting access to some subareas for specific groups.
It's impracticable to grant access to every openly accessible subarea in the global group and adding further `allow`-rules to groups respectively.

Your help is appreciated!

# Example
### Permissions
Everything in the _Content_-column
### Pagerules
- **Allow** `read:pages` `write:pages` `manage:pages` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/`
- **Deny** everything on **starts with** `/subarea`
### Permissions
`read:pages` `write:pages` `manage:pages` `read:source` `read:history` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` 
### Pagerules
- **Allow** `read:pages` `write:pages` `manage:pages` `read:source` `read:history` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/subarea`

**Comment by rkone** [comment]
I read the docs as this:

- **Allow** `read:pages` `write:pages` `manage:pages` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/`
- **Allow** (nothing) on **starts with** `/subarea`

Since the `/subarea` rule is more specific, it should override the `/` rule. Since the default is deny, and the `/subarea` rule does not allow anything, it should deny everything.  **It does not.**

My workaround is:
- **Allow** `read:pages` `write:pages` `manage:pages` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/`
- **Deny**  `read:pages` `write:pages` `manage:pages` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/subare` 

Notice the missing "a" on the deny rule. Then in subarea group I have the following rule:
- **Allow** `read:pages` `write:pages` `manage:pages` `read:assets` `write:assets` `manage:assets` `read:comments` `write:comments` on **starts with** `/subarea` (with final a)

**But this is a hack.** Any page that was created in a folder for `/subarenotworking` will not give the permissions expected!
